--- a/docstring_to_markdown/__init__.py
+++ b/docstring_to_markdown/__init__.py
@@ -1,10 +1,10 @@
-from importlib_metadata import entry_points
+from importlib.metadata import entry_points
 from typing import List, TYPE_CHECKING
 
 from .types import Converter
 
 if TYPE_CHECKING:
-    from importlib_metadata import EntryPoint
+    from importlib.metadata import EntryPoint
 
 __version__ = "0.17"
 
