--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,9 +21,9 @@
 
 cmake_minimum_required(VERSION 3.12.0)
 
-# Try C++14, then fall back to C++11 and C++98.  Used for feature tests
+# Try C++17, then fall back to the older standards.  Used for feature tests
 # for optional features.
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 # Use folders (for IDE project grouping)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
