--- a/locate/Makefile.am
+++ b/locate/Makefile.am
@@ -46,7 +46,7 @@
 	rm -f $@
 	find=`echo find|sed '$(transform)'`; \
 	frcode=`echo frcode|sed '$(transform)'`; \
-	copyright=`sed -n '/^# Copyright /{s/^..//;p;q;}' $(srcdir)/updatedb.sh \
+	copyright=`sed -n '/^# Copyright /{s/^..//;p;q;}' $(srcdir)/updatedb.sh | \
 	  grep .` || exit 1; \
 	sed \
 	-e "s,@""bindir""@,$(bindir)," \
--- a/locate/Makefile.in
+++ b/locate/Makefile.in
@@ -3246,7 +3246,7 @@
 	rm -f $@
 	find=`echo find|sed '$(transform)'`; \
 	frcode=`echo frcode|sed '$(transform)'`; \
-	copyright=`sed -n '/^# Copyright /{s/^..//;p;q;}' $(srcdir)/updatedb.sh \
+	copyright=`sed -n '/^# Copyright /{s/^..//;p;q;}' $(srcdir)/updatedb.sh | \
 	  grep .` || exit 1; \
 	sed \
 	-e "s,@""bindir""@,$(bindir)," \
