--- a/auxiliary_files/mumble-server.ini	2026-09-28 03:09:42.489150811 +0530
+++ b/auxiliary_files/mumble-server.ini	2026-09-28 03:09:42.506355772 +0530
@@ -13,7 +13,7 @@
 
 ; Path to database. If blank, will search for
 ; mumble-server.sqlite in default locations or create it if not found.
-database=
+database=/var/lib/mumble-server/mumble-server.sqlite
 
 ; The server defaults to using SQLite with its default rollback journal.
 ; In some situations, using SQLite's write-ahead log (WAL) can be
