diff -ru a/src/hal/components/Submakefile b/src/hal/components/Submakefile
--- a/src/hal/components/Submakefile
+++ b/src/hal/components/Submakefile
@@ -32,8 +32,7 @@
 $(COMP_MANPAGES): ../docs/man/man9/%.9: hal/components/%.comp ../bin/halcompile
 	$(ECHO) Making halcompile manpage $(notdir $@)
 	@mkdir -p $(dir $@)
-	$(Q)../bin/halcompile -U --document -o $@.new $< && preconv -r < $@.new > $@
-	$(RM) $@.new
+	$(Q)../bin/halcompile -U --document -o $@ $<
 
 $(COMP_DRIVER_MANPAGES): ../docs/man/man9/%.9: hal/drivers/%.comp ../bin/halcompile
 	$(ECHO) Making halcompile manpage $(notdir $@)
