--- a/gui/include/gui/dychart.h
+++ b/gui/include/gui/dychart.h
@@ -122,13 +122,6 @@
 #define _putenv putenv  // This is for other Windows compiler
 #endif
 
-//----------------------------------------------------------------------------
-//              Use the CPL Portability library only if S57 is enabled
-//----------------------------------------------------------------------------
-
-#define USE_CPL
-// #include "gdal/cpl_port.h"
-
 #ifndef NULL
 #define NULL 0
 #endif
