diff -ru a/src/qt5ct-qtplugin/main.cpp b/src/qt5ct-qtplugin/main.cpp
--- a/src/qt5ct-qtplugin/main.cpp	2025-01-27 23:46:40.000000000 +0530
+++ b/src/qt5ct-qtplugin/main.cpp	2026-09-28 15:39:48.378929278 +0530
@@ -42,7 +42,8 @@
 QPlatformTheme *Qt5CTPlatformThemePlugin::create(const QString &key, const QStringList &params)
 {
     Q_UNUSED(params);
-    if (key.toLower() == "qt5ct" || key.toLower() == "qt6ct")
+    if (key.toLower() == "qt5ct" || key.toLower() == "qt6ct" ||
+        key.toLower() == "kde")
         return new Qt5CTPlatformTheme();
 
     return nullptr;
diff -ru a/src/qt5ct-qtplugin/qt5ct.json b/src/qt5ct-qtplugin/qt5ct.json
--- a/src/qt5ct-qtplugin/qt5ct.json	2025-01-27 23:46:40.000000000 +0530
+++ b/src/qt5ct-qtplugin/qt5ct.json	2026-09-28 15:39:48.378801475 +0530
@@ -1,3 +1,3 @@
 {
-    "Keys": [ "qt5ct", "qt6ct" ]
+    "Keys": [ "qt5ct", "qt6ct", "kde" ]
 }
