diff -ru a/src/mainwindow.cpp b/src/mainwindow.cpp
--- a/src/mainwindow.cpp	2026-09-28 03:57:47.100420671 +0530
+++ b/src/mainwindow.cpp	2026-09-28 03:57:47.149906172 +0530
@@ -282,7 +282,7 @@
 
     InValidCharsInPrevCall = false;
 
-    checkNewVersions = true;
+    checkNewVersions = false;
     reportInfo = false;
     configured = false;
 
@@ -7213,7 +7213,7 @@
     manageMode = (settings.value ("IncludeModeForNeeded", true).toBool ());
     awardsWidget->setIncludeModeForNeeded (manageMode);
     searchWidget->setShowCallInSearch(settings.value ("ShowCallsignInSearch", true).toBool ());
-    checkNewVersions = settings.value ("CheckNewVersions", true).toBool ();
+    checkNewVersions = settings.value ("CheckNewVersions", false).toBool ();
     reportInfo = false;
     alwaysADIF = settings.value ("AlwaysADIF", true).toBool ();
     setLogLevel(util->stringToDebugLevel(settings.value ("DebugLog").toString ()));
diff -ru a/src/setuppages/setuppagemisc.cpp b/src/setuppages/setuppagemisc.cpp
--- a/src/setuppages/setuppagemisc.cpp	2026-09-28 03:57:47.117164685 +0530
+++ b/src/setuppages/setuppagemisc.cpp	2026-09-28 03:57:47.151255810 +0530
@@ -662,7 +662,7 @@
     sendQSLWhenRecCheckBox->setChecked (settings.value("SendQSLWhenRec", true).toBool ());
     showStationCallWhenSearchCheckBox->setChecked (settings.value("ShowCallsignInSearch", true).toBool ());
     completeWithPreviousCheckBox->setChecked (settings.value("CompleteWithPrevious").toBool ());
-    checkNewVersionCheckBox->setChecked (settings.value("CheckNewVersions", true).toBool ());
+    checkNewVersionCheckBox->setChecked (settings.value("CheckNewVersions", false).toBool ());
     useDxMarathonCheckBox->setChecked (settings.value("ManageDXMarathon").toBool ());
     sendEQSLByDefaultSearchCheckBox->setChecked (settings.value("SendEQSLByDefault", true).toBool ());
     deleteAlwaysAdiFileCheckBox->setChecked (settings.value("DeleteAlwaysAdiFile").toBool ());
